solidity-plugin/src/main/java/org/sonarsource/solidity/CognitiveComplexityVisitor.java

```diff
@@ -27,55 +29,85 @@ public CognitiveComplexityVisitor(SourceUnitContext sourceUnitCtx) {
   @Override
   public Token visitFunctionDefinition(FunctionDefinitionContext ctx) {
     complexity = 0;
-    System.out.println("Starts counting..." + ctx.getText() + " --- " + complexity);
+    nestingLevel = 0;
+    System.out.println("Starts counting..." + ctx.getText() + " --- complexity: " + complexity + " --- level: " + nestingLevel);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=20&issues=AWQdvLOV_bEzQ-kUUpm6&open=AWQdvLOV_bEzQ-kUUpm6)

[ISSUE_KEY]: AWQdvLOV_bEzQ-kUUpm6